configure.py

```diff
@@ -2982,6 +2994,37 @@ def canonicalize_options(options, info_os, info_arch):
     if options.cpu != options.arch:
         logging.info('Canonicalized CPU target %s to %s', options.cpu, options.arch)
 
+    # select and sanity check build targets
+    def canonicalize_build_targets(options):
+        build_targets = read_build_targets(options)
+
+        # --build-targets was not provided: build default targets
+        if build_targets is None:
+            options.build_targets = "cli,tests"
+            return
+
+        # --build-targets was set to empty string: invalid input
+        if not build_targets:
+            raise UserError("select at least one --build-targets")
+
+        for build_target in build_targets:
+            if build_target not in ACCEPTABLE_BUILD_TARGETS:
+                raise UserError("unknown build target: %s" % build_target)
+
```

Comment:
I think it is possible here for a user on MSVC to request static+shared which doesn't work due to how Windows DLLs work. It should probably error instead.

Reply:
Thats true. If a user calls `./configure --build-targets="static,shared` on windows, she will be presented with an error due to a sanity check that is already part of the script (see [here](https://github.com/randombit/botan/blob/0ee18280a798c3f311259036ae70e503ae9ff3a3/configure.py#L3057).

In fact, I ran into this issue [in the CI](https://ci.appveyor.com/project/randombit/botan/builds/27287793/job/g8g2l2yki2wc8cc3) once as well, before adding https://github.com/randombit/botan/pull/2098/commits/577ac1a01b8b3ead65655de5ef84176253549428.

Reply:
True. However, as stated in a different reply, this will be caught just a few lines below. Admittedly, this is somewhat hard to spot, as `canonicalize_build_targets` sets `options.build_static_lib` and `options.build_shared_lib` accordingly.

Reply:
Ah sorry I missed this, what's there is fine then.